### ISSUE TITLE
Add __str__ method for FactorSet

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#553]
Current behavior:

``` python
In [1]: from pgmpy.factors import Factor
In [2]: from pgmpy.factors import FactorSet
In [3]: phi1 = Factor(['x1', 'x2', 'x3'], [2, 3, 2], range(12))
In [4]: phi2 = Factor(['x3', 'x4', 'x1'], [2, 2, 2], range(8))
In [5]: factor_set1 = FactorSet(phi1, phi2)
In [6]: print(factor_set1)
<pgmpy.factors.FactorSet.FactorSet object at 0x7fc1a599e940
```

Expected behavior:

``` python
In [ ]: print(factor_set1)
Out[ ]: [<Factor representing phi(x1:2, x2:3, x3:2) at 0x7fc1a599e4a8,
         <Factor representing phi(x3:2, x4:2, x1:2) at 0x7fc1a599e470]
```

**Comment by kshitij10496** [comment]
I am working on this.

**Comment by khalibartan** [comment]
@kshitij10496 I'll work on this, since I'm going through the things. I have already opened a PR , in which I'll cover up small issues .

**Comment by kshitij10496** [comment]
Oh cool
Go on :+1:

**Comment by khalibartan** [comment]
@kshitij10496 Any other issues , which You want me to take a look at?

**Comment by khalibartan** [comment]
@ankurankan We want this expected behaviour or should I print the `__str__` representation of factors in the factor set.
This:

```
In [1]: from pgmpy.factors import Factor

In [2]: from pgmpy.factors import FactorSet

In [3]: phi1 = Factor(['x2','x1','x3'],[3,2,2], range(12))

In [4]: phi2 = Factor(['x3','x4','x1'],[2,2,2], range(8))

In [5]: factor_set = FactorSet(phi1,phi2)

In [7]: print(factor_set)  
+------+------+------+-----------------+
| x2   | x1   | x3   |   phi(x2,x1,x3) |
|------+------+------+-----------------|
| x2_0 | x1_0 | x3_0 |          0.0000 |
| x2_0 | x1_0 | x3_1 |          1.0000 |
| x2_0 | x1_1 | x3_0 |          2.0000 |
| x2_0 | x1_1 | x3_1 |          3.0000 |
| x2_1 | x1_0 | x3_0 |          4.0000 |
| x2_1 | x1_0 | x3_1 |          5.0000 |
| x2_1 | x1_1 | x3_0 |          6.0000 |
| x2_1 | x1_1 | x3_1 |          7.0000 |
| x2_2 | x1_0 | x3_0 |          8.0000 |
| x2_2 | x1_0 | x3_1 |          9.0000 |
| x2_2 | x1_1 | x3_0 |         10.0000 |
| x2_2 | x1_1 | x3_1 |         11.0000 |
+------+------+------+-----------------+
+------+------+------+-----------------+
| x3   | x4   | x1   |   phi(x3,x4,x1) |
|------+------+------+-----------------|
| x3_0 | x4_0 | x1_0 |          0.0000 |
| x3_0 | x4_0 | x1_1 |          1.0000 |
| x3_0 | x4_1 | x1_0 |          2.0000 |
| x3_0 | x4_1 | x1_1 |          3.0000 |
| x3_1 | x4_0 | x1_0 |          4.0000 |
| x3_1 | x4_0 | x1_1 |          5.0000 |
| x3_1 | x4_1 | x1_0 |          6.0000 |
| x3_1 | x4_1 | x1_1 |          7.0000 |
+------+------+------+-----------------+
```

Or this

```
In [8]: print(factor_set)
Out[8]: 
{<Factor representing phi(x2:3, x1:2, x3:2) at 0x7fddd01cd490>,
 <Factor representing phi(x3:2, x4:2, x1:2) at 0x7fddd01cd4d0>}
```

**Comment by ankurankan** [comment]
@khalibartan The second one.

**Comment by khalibartan** [comment]
@ankurankan I think this should do

```
In [1]: %paste
        >>> from pgmpy.factors import FactorSet
        >>> from pgmpy.factors import Factor
        >>> phi1 = Factor(['x1', 'x2', 'x3'], [2, 3, 2], range(12))
        >>> phi2 = Factor(['x3', 'x4', 'x1'], [2, 2, 2], range(8))
        >>> factor_set = FactorSet(phi1, phi2)

## -- End pasted text --

In [2]: print(factor_set)
set([<Factor representing phi(x1:2, x2:3, x3:2) at 0x7f001f964ed0>, <Factor representing phi(x3:2, x4:2, x1:2) at 0x7f001f976150>])
```

the output is similar to printing a set.
Will be fixed in #552

**Comment by khalibartan** [comment]
@ankurankan Close this , Fixed in #552
